pandas/_libs/testing.pyx

```diff
@@ -73,31 +68,33 @@ cpdef assert_almost_equal(a, b,
     ----------
     a : object
     b : object
-    check_less_precise : bool or int, default False
```

Comment:
leave this in here and add a deprecated tag

Reply:
you can change the default to Noen and interpret it inside

Reply:
its fine to change all internal usages, but leave at least 1 test for any external usages (that tests the deprecation)

Reply:
I'm not accustomed to editing `pyx` code. Would it be something like this:

```python

    # TODO: `check_less_precise` is deprecated. Remove this bit, when it
    #       is finally removed.
    if isinstance(check_less_precise, bool):
        if check_less_precise:
            # 3-digit tolerance
            rtol = atol = 0.5e-3
        else:
            # 5-digit tolerance
            rtol = atol = 0.5e-5
    elif instance(check_less_precise, int):
        # Equivalent to setting checking_less_precise=<decimals>
        rtol = atol = 0.5 * 10 ** -check_less_precise

```
?

Also, when talking about tests, you mean literally implementing a test that checks on the call to this function (`_libs.testing.assert_almost_equal`), right?

Reply:
> Also, when talking about tests, you mean literally implementing a test that checks on the call to this function (_libs.testing.assert_almost_equal), right?

yes.  this would go in pandas.tests.util.test_something_or_other and would look something like:

```
def test_old_version_deprecated():    #  <-- fill in a more informative name
    # GH#30562
    with tm.assert_produces_warning(FutureWarning):
         relevant_function(old_args, old_kwargs)
```